lib/rules/landmark-unique-matches.js

```diff
@@ -0,0 +1,39 @@
+const excludedParentsForHeaderFooterLandmarks = [
+	'article',
+	'aside',
+	'main',
+	'nav',
+	'section'
+].join(',');
+function isHeaderFooterLandmark(headerFooterElement) {
+	return !axe.commons.dom.findUp(
+		headerFooterElement,
+		excludedParentsForHeaderFooterLandmarks
+	);
+}
+
+function isLandmarkVirtual({ actualNode }) {
+	const landmarkRoles = axe.commons.aria.getRolesByType('landmark');
+	const role = axe.commons.aria.getRole(actualNode);
+	const tagName = actualNode.nodeName.toLowerCase();
+
+	if (tagName === 'header' || tagName === 'footer') {
+		return isHeaderFooterLandmark(actualNode);
+	}
+
+	if (tagName === 'section' || tagName === 'form') {
+		const label = axe.commons.aria.label(actualNode);
+		return !!label;
+	}
+
+	return (role && landmarkRoles.indexOf(role) >= 0) || role === 'region';
+}
+
```

Comment:
Replace from here on down with

```
return (
    isLandmarkVirtual(virtualNode) && axe.commons.dom.isVisible(node, true)
);
```

Reply:
Done.

Reply:
@waabid I am still showing the old code

Reply:
Sorry; I think I misunderstood you. I assume the new changes are what you're looking for. 😄